pkg/controller/dnsendpoint/dnsendpoint_controller.go

```diff
@@ -145,10 +179,19 @@ func (r *ReconcileDNSEndpoint) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, nil
 	}
 
-	domain := instance.Spec.Zone
-	dnsLog = dnsLog.WithField("domain", domain)
+	fullDomain := instance.Spec.Zone
+	splitDomain := strings.Split(fullDomain, ".")
+	rootDomain := strings.Join(splitDomain[1:], ".")
+
+	dnsLog = dnsLog.WithField("domain", fullDomain)
 
-	if !r.nameServerScraper.HasBeenScraped(domain) {
+	scraper, ok := r.nameServerScraperMap[rootDomain]
+	if !ok {
+		dnsLog.Warn("no scraper for domain found, skipping reconcile")
```

Comment:
This should be an error-level log entry. It is a failure of the validation to get a domain that is not covered by one of the managed domains.